### PR TITLE
Composite measures dashboard

### PR DESCRIPTION
**Pull request by lukoktonos**
This PR:
- Adds Front-end support for composite measures on dashboard page
- Adds support for exporting front-end and back-end composite measure fixtures
- Adds fixture files for composite measure

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1753
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable) (new front-end tests pass, there are back-end tests failing on base branch)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) ) https://jira.mitre.org/browse/BONNIE-1792 will finish this process once it's possible (after deployment).
- [x] Code coverage has not gone down and all code touched or added is covered. 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
composite_measures | 2162 / 2454 LOC (88.1%) covered. | Statements   : 59.57% ( 7306/12264 )<br>Branches     : 43.54% ( 1990/4570 )<br>Functions    : 49.98% ( 1210/2421 )<br>Lines        : 57.47% ( 6562/11418 )
composite_measures_dashboard| 2162 / 2454 LOC (88.1%) covered. | Statements   : 59.62% ( 7322/12281 )<br>Branches     : 43.55% ( 1991/4572 )<br>Functions    : 50.12% ( 1217/2428 )<br>Lines        : 57.53% ( 6578/11435 )
- [x] Automated regression test(s) pass N/A no changes affect calculation.  Regression test updated to support new view in https://gitlab.mitre.org/bonnie/server-scripts/merge_requests/20

**Reviewer 1:**

Name: @hackrm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: app/assets/javascripts/models/measure.js.coffee [hunk 1]
```diff
@@ -129,6 +135,9 @@ class Thorax.Models.Measure extends Thorax.Model
     # return field values sorted by title
     _(fields).sortBy (field) -> field.title
 
+  isComponentMeasure: ->
+    /&/.test @get('hqmf_set_id')
```
Comment (jbradl11) [inline app/assets/javascripts/models/measure.js.coffee]: Is there another way to test if the measure is a component measure? I am concerned about relying on a special character.

Comment (lukoktonos) [reply app/assets/javascripts/models/measure.js.coffee]: I don't believe so.  Bonnie bundler currently uses this method for detection as well.

Comment (p9g) [reply app/assets/javascripts/models/measure.js.coffee]: HQMF for composite is required to have
```
<subjectOf>
  <measureAttribute>
    <code code="CMPMSRMTH" codeSystem="2.16.840.1.113883.5.4"/>
    <value code="{code}" codeSystem="2.16.840.1.113883.5.1063" xsi:type="CD"/>
  </measureAttribute>
</subjectOf>
```
Theoretically, a composite can have components which are also composite measures.
But if the subjectOf/measureAttribute/code element above is not in the HQMF, it is not a composite.

Comment (lukoktonos) [reply app/assets/javascripts/models/measure.js.coffee]: @p9g We have a field on the measure to indicate whether it is a `composite` or not.  This function is to determine if a measure is a `component` of a `composite`.

Comment (jbradl11) [reply app/assets/javascripts/models/measure.js.coffee]: There is update to Bonnie_Bundler so that there will be `composite` and `component` flags on the measure model.

**Comment by jbradl11** [comment]
This looks good. I think an addition to this task could be to disable "Delete" and "Update" buttons in the "Gear" button on the component measure details pages. Let me know if you disagree.

**Review by jbradl11 — APPROVED** [review]
Looks good!

**Review by mayerm94 — APPROVED** [review]
Works great
